src/snowflake/snowpark/mock/_functions.py

```diff
@@ -609,120 +607,174 @@ def _to_timestamp(
     fmt: Optional[ColumnEmulator],
     try_cast: bool = False,
     add_timezone: bool = False,
+    caller=None,
```

Comment:
NIT: instead of caller, what about a boolean variable like: `enforce_ltz`

Reply:
I'm good with renaming it